doc/experiment.rst

```diff
@@ -1,326 +1,350 @@
+
+Initializing
+============
+.. compound::
+  To create a workflow, we *initialize* an ``Experiment`` object
+  once at the beginning of the Python driver script.
+  To create an ``Experiment``, we specify a name
+  and the system launcher with which all entities will be executed.
+  Here, we are running the example on a Slurm machine and will
+  set the `launcher` argument to `auto` to instruct SmartSim to
+  attempt to find the WLM.
+
+  .. literalinclude:: ../tutorials/doc_examples/experiment_doc_examples/exp.py
+    :language: python
+    :linenos:
+    :lines: 1-7
+
+  We also initialize a SmartSim logger. We will use the logger to log the ``Experiment``
+  summary.
+
+.. compound::
+  Next, we will launch a SmartSim in-memory database called an ``Orchestrator``.
+  To *initialize* an ``Orchestrator`` object, use the ``Experiment.create_database()``
+  function. We will create a multi-sharded ``Orchestrator`` and therefore will set
+  the argument `db_nodes` to 3. SmartSim will assign a `port` to the ``Orchestrator``
+  and attempt to detect your machine's interface if values are not provided to the
+  ``Experiment.create_database()`` factory method.
+
+  .. literalinclude:: ../tutorials/doc_examples/experiment_doc_examples/exp.py
+    :language: python
+    :linenos:
+    :lines: 9-10
+
+.. compound::
+  Before invoking the factory method to create a ``Model``, we must
+  first create a ``RunSettings`` object which holds the information needed to execute the ``Model``
+  on the system. The ``RunSettings`` object is initialized using the
+  ``Experiment.create_run_settings()`` factory method. In this factory method,
+  we specify the executable to run and the arguments to pass to
+  the executable.
+
+  The example ``Model`` is a simple `Hello World` program
+  that echos `Hello World` to stdout.
+
+  .. literalinclude:: ../tutorials/doc_examples/experiment_doc_examples/exp.py
+    :language: python
+    :linenos:
+    :lines: 12-15
+
+  After creating the ``RunSettings`` object, the ``Model`` object can be created and initialized using
+  the ``RunSettings`` object via the ``Experiment.create_model()`` function. In the ``Model`` factory method,
+  the ``Model`` `name` and the ``RunSettings`` object are provided as input parameters.
+
+Generating
+==========
+.. compound::
+  Next we generate the file structure for the ``Experiment``. A call to ``Experiment.generate()``
+  instructs SmartSim to create directories within the ``Experiment`` folder for each instance passed in.
+  We plan to organize the ``Orchestrator`` and ``Model`` output files within the ``Experiment`` folder and
+  therefore pass the ``Orchestrator`` and ``Model`` instances to ``exp.generate()``:
+
+  .. literalinclude:: ../tutorials/doc_examples/experiment_doc_examples/exp.py
+    :language: python
+    :linenos:
+    :lines: 17-18
+
+  `Overwrite=True` instructs SmartSim to overwrite each file contents if the ``Experiment`` is
```

Comment:
if the ``Experiment`` is executed again. -> if entity files and directories already exist within the experiment directory.